client/src/pages/ViewRequestStatus/RequestStatusView.js

```diff
@@ -30,6 +49,11 @@ export function RequestStatus() {
 					setDonation(laptopDonation);
 				});
 		}
+		if (request != null && request.status == "WAITING") {
```

Comment:
This is wrong behaviour so I don't think we can merge it: Move the notification to the event handler for laptop_request:statusChanged, saying something like "Good news, we've found you a laptop!"